responsibleai/tests/rai_insights/test_rai_insights_validations.py

```diff
@@ -532,6 +531,150 @@ def test_misidentified_categorical_features(self):
+                         columns=['c1', 'c2'])
+        y = np.array([1, 0])
+        model = create_lightgbm_classifier(X, y)
+
+        X[TARGET] = y
+        feature_metadata = FeatureMetadata(
+            categorical_features=feature_metadata_categorical_features)
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match='The categorical_features provided via the '
+                      'RAIInsights constructor and the categorical_features '
+                      'provided via the feature_metadata argument do not '
+                      'match.'):
+            RAIInsights(
+                model=model,
+                train=X,
+                test=X,
+                target_column=TARGET,
+                task_type='classification',
+                categorical_features=arg_categorical_features,
+                feature_metadata=feature_metadata)
+
+    @pytest.mark.parametrize(
+        'categorical_features',
+        [None, ['c1'], ['c1', 'c2']])
+    def test_feature_metadata_categorical_features_only(
+            self, categorical_features):
+        X = pd.DataFrame(data=[[1, 1], [2, 3]],
+                         columns=['c1', 'c2'])
+        y = np.array([1, 0])
+        model = create_lightgbm_classifier(X, y)
+
+        X[TARGET] = y
+        feature_metadata = FeatureMetadata(
+            categorical_features=categorical_features)
+
+        RAIInsights(
+            model=model,
+            train=X,
+            test=X,
+            target_column=TARGET,
+            task_type='classification',
+            feature_metadata=feature_metadata)
+
+    @pytest.mark.parametrize("feature_metadata", [
+        FeatureMetadata(datetime_features=['c1']),
+        FeatureMetadata(time_series_id_features=['c1'])
+    ])
+    def test_feature_metadata_unsupported_time_series_features(
+            self, feature_metadata):
+        X = pd.DataFrame(data=[[1, 1], [2, 3]],
+                         columns=['c1', 'c2'])
+        y = np.array([1, 0])
+        model = create_lightgbm_classifier(X, y)
+        X[TARGET] = y
+
+        changed_metadata_field = [
+            k for k in feature_metadata.__dict__.keys()
+            if feature_metadata.__dict__[k] is not None][0]
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match=f"The specified metadata {changed_metadata_field} is "
+                      "only supported for the forecasting task type."):
+            RAIInsights(
+                model=model,
+                train=X,
+                test=X,
+                target_column=TARGET,
+                task_type='classification',
+                feature_metadata=feature_metadata)
+
+    @pytest.mark.skip(
+        "Skip forecasting validation test until forecasting is enabled.")
+    def test_feature_metadata_forecasting_multiple_datetime_features(self):
+        X = pd.DataFrame(data=[[1, 1], [2, 3]],
```

Comment:
This seems to be a repeated code of creating a dataframe. Maybe move it into a function in common_utils.py or create a fixture.

Reply:
Isn't it way easier to read and understand the test case if we have it right there? Otherwise I have to find the definition of the X variable somewhere else for such a simple data frame that is of no particular significance to this test case or any other. It's 1 line of code regardless.